br/pkg/storage/azblob.go

```diff
@@ -380,6 +380,25 @@ func (s *AzureBlobStorage) Create(ctx context.Context, name string) (ExternalFil
 	return uploaderWriter, nil
 }
 
+func (s *AzureBlobStorage) Rename(ctx context.Context, oldFileName, newFileName string) error {
+	data, err := s.ReadFile(ctx, oldFileName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = s.WriteFile(ctx, newFileName, data)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return s.DeleteFile(ctx, oldFileName)
+}
+
+// AtomicWriteFile implements ExternalStorage interface.
+func (s *AzureBlobStorage) AtomicWriteFile(ctx context.Context, name string, data []byte) error {
+	// AzureBlobStorage WriteFile use interface about BlockBlob Upload which may be atomic
```

Comment:
can we ensure the operation is atomic like aws s3

Reply:
i think we can ensure,  `WriteFile` eventually call `Upload`, and [upload](https://pkg.go.dev/github.com/Azure/azure-storage-blob-go/azblob#BlockBlobURL.Upload) is atomic.